testing_app/test_driver/perf_test.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Testing App Performance Tests', () {
+    FlutterDriver driver;
+
+    // Connect to the Flutter driver before running any tests.
+    setUpAll(() async {
+      driver = await FlutterDriver.connect();
+    });
+
+    // Close the connection to the driver after the tests have completed.
+    tearDownAll(() async {
+      if (driver != null) {
+        await driver.close();
+      }
+    });
+
+    test('Scrolling test', () async {
+      // Create Finders that are used multiple times.
+      final listFinder = find.byType('ListView');
+      final firstItem = find.byValueKey('text_0');
+      final lastItem = find.byValueKey('text_49');
+
+      // Record a performance profile as the app scrolls through
+      // the list of items.
+      final scrollingTimeline = await driver.traceAction(() async {
+        // Scroll until the item to be found appears.
+        // Use dxScroll to scroll horizontally and dyScroll
+        // to scroll vertically.
+        await driver.scrollUntilVisible(
+          listFinder,
+          lastItem,
+          dyScroll: -1000.0,
+        );
+
+        // Check if the item contains the correct text.
+        expect(await driver.getText(lastItem), 'Item 49');
+
+        // Scroll back up all the way.
+        await driver.scrollUntilVisible(
+          listFinder,
+          firstItem,
+          dyScroll: 1000.0,
```

Comment:
I'm not sure about this `1000.0` number (so is the `-1000` above): this seems too large and might miss the target in a device with smaller screen. The typical value we are using in devicelab tests are 20~50.

Reply:
I can surely reduce that. But, we are trying to test the scrolling performance by scrolling all the way down and then back up. Don't you think 20~50 will be too slow to test the exact performance in this case? Please correct me if I'm wrong.

Reply:
If that's the case, you probably don't want to use `driver.scrollUntilVisible` but `driver.scroll`,  and your list looks too short not exceeding `ListView .cacheExtent`

Reply:
Thanks for correcting me! I was trying to use `driver.scroll()` when I had written the test, but for some reason, I wasn't able to achieve my objective at that time.
Now, this is what I've changed in the code:
- Increased `itemCount` from `50` to `100`
- Set `cacheExtent` to `20`
- Used `scroll()` to scroll down and back up with dy : `7000` and `-7000` and duration of `1` second

I'm able to achieve my goal using this approach.

If that seems good, I'll push the changes soon.